src/App.tsx

```diff
@@ -1,14 +1,73 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.scss';
+import debounce from 'lodash.debounce';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { PeopleList } from './components/PeopleList';
+
+type FilterParams = {
+  query: string;
+};
+
+function filterPeople(people: Person[], { query }: FilterParams): Person[] {
+  let resultPeople = [...people];
+  const normalazidQuery = query.toLowerCase();
+
+  if (query) {
+    resultPeople = resultPeople.filter((person) => {
+      const normalazidName = person.name.toLowerCase();
+
+      return normalazidName.includes(normalazidQuery);
+    });
+  }
+
+  return resultPeople;
+}
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [preparedQuery, setPreparedQueary] = useState('');
+  const setQueryToFilter = useCallback(
+    debounce(setPreparedQueary, 1000),
```

Comment:
Debounce must be customizable, so it better to use variable for these

Reply:
+ here as well